File: examples/dotnet/AsyncTaskPolling.cs
using AILabTools;
using AILabTools.Generated.Params;
using AILabTools.Http;

using var client = new AILabToolsClient(
    Environment.GetEnvironmentVariable("AILAB_API_KEY")!);

var task = await client.Portrait.ChangeHairstyleAsync(
    new PortraitHairstyleEditingProParams
    {
        TaskType = "async",
        Image = FileInput.FromPath("./image.jpg"),
        HairStyle = "BuzzCut",
    });
var taskId = task.TaskId ?? task.Data?.TaskId
    ?? throw new InvalidOperationException("The API did not return a task ID.");

var result = await client.WaitForTaskAsync(taskId);
Console.WriteLine(result.Data?.TaskStatus);

File: examples/swift/AsyncTaskPolling.swift
import AILabTools
import Foundation

let client = try AILabToolsClient(
    apiKey: ProcessInfo.processInfo.environment["AILAB_API_KEY"]!
)

let task = try await client.portrait.changeHairstyle(
    PortraitHairstyleEditingProParams(
        taskType: "async",
        image: .fileURL(URL(fileURLWithPath: "./image.jpg")),
        hairStyle: "BuzzCut"
    )
)
guard let taskId = task.taskId ?? task.data?.taskId else {
    throw AILabToolsError.invalidResponse("The API did not return a task ID.")
}

let result = try await client.waitForTask(taskId)
print(result.data?.taskStatus ?? -1)
client.close()
